cpp/arcticdb/entity/type_utils.hpp

```diff
@@ -12,8 +12,13 @@
 namespace arcticdb {
 
 inline bool trivially_compatible_types(entity::TypeDescriptor left, entity::TypeDescriptor right) {
-    if(left == right)
+    if(left == right) {
```

Comment:
While I don't want to enforce my personal preferences I strongly believe we need some sort of coding standard for this. Homogeneous code is easier to read. Currently if different people touch the same function it's possible to end up with this:
```c++
if(condition) return true;
...
if(other_condition)
    return true;
...
if(third_condition) {
    return true;
}
```

Reply:
Here is a snippet from [our code](https://github.com/man-group/ArcticDB/blob/d6b8c2a0eb00163942c6e97c7d10a628e00bbbbf/cpp/arcticdb/column_store/memory_segment_impl.hpp#L287)

```c++
                if constexpr (T::DimensionTag::value == Dimension::Dim0) {
                    if constexpr (is_sequence_type(T::DataTypeTag::data_type)) {
                        // test only for now
                        throw std::runtime_error("string type not implemented");
                    } else {
                        if constexpr(std::is_same_v<RawType, S>)
                        val = that->parent_->scalar_at<RawType>(that->row_id_, col);
                        else
                            util::raise_rte("Type mismatch in scalar access");
                    }
                } else {
                    throw std::runtime_error("Scalar method called on multidimensional column");
                }
```
Here we have if/else wrapped in block, if else without block and to spice things up different indentation of the codepaths

Reply:
The indentation is just plain wrong, I would always fix this where I noticed it. Other than that the bracketing does follow a pattern, which is that where both if and else blocks are a single line it is permitted to omit the braces, otherwise they should be used.

One of the things that I think aids readibility is not having too much vertical sprawl. I really don't want to put braces around every one-line branch for that reason.